core/TypePtr.cc

```diff
@@ -91,4 +91,325 @@ bool TypePtr::isBottom() const {
     return t != nullptr && t->symbol == Symbols::bottom();
 }
 
+int TypePtr::kind() const {
+    switch (tag()) {
+        case Tag::AppliedType:
+            return 1;
+        case Tag::BlamedUntyped:
+        case Tag::UnresolvedAppliedType:
+        case Tag::UnresolvedClassType:
+        case Tag::ClassType:
+            return 2;
+        case Tag::LiteralType:
+            return 3;
+        case Tag::ShapeType:
+            return 4;
+        case Tag::TupleType:
+            return 5;
+        case Tag::LambdaParam:
+        case Tag::SelfTypeParam:
+            return 6;
```

Comment:
It's unfortunate how close `kind()` is to `tag()`.

Reply:
If there weren't ties I'd just align the enum type to match `kind`. But there are ties, unfortunately.